cmd/install.go

```diff
@@ -108,19 +94,61 @@ func getUserList(sysUsers v1beta1.SystemUser) []string {
 }
 
 func getCmdArgs(role string) []string {
+	params := make(map[string]string)
+
+	params["data-dir"] = dataDir
+
 	var args []string
 
 	if role == "server" {
 		args = append(args, "server")
+
+		//server params
+		if serverWorkerProfile != "default" {
+			params["profile"] = serverWorkerProfile
+		}
+
+		if enableWorker {
+			params["enable-worker"] = ""
+		}
+
+		params["token-file"] = serverToken
+
 		if cfgFile != "" {
 			args = append(args, "--config", cfgFile)
 		}
 	} else {
-		args = append(args, "worker", "--token-file", "REPLACEME")
+		args = append(args, "worker")
+
+		//worker params
+		if workerProfile != "default" {
+			params["profile"] = workerProfile
+		}
+
+		if tokenFile != "" {
+			params["token-file"] = tokenFile
+		} else {
+			params["token-file"] = "REPLACEME"
+		}
+		if cloudProvider {
+			params["enable-cloud-provider"] = ""
+		}
+		params["cri-socket"] = criSocket
 	}
 
 	if debug {
 		args = append(args, "--debug")
 	}
+
+	for k, v := range params {
+		if v != "" {
+			args = append(args, (fmt.Sprintf("--%s %s", k, v)))
+		}
+
+		if k == "enable-worker" || k == "enable-cloud-provider" {
+			args = append(args, (fmt.Sprintf("--%s", k)))
+
+		}
+	}
```

Comment:
```suggestion
	for k, v := range params {
		arg := "--" + k
		if v != "" {
			arg += "=" + v
		}
		args = append(args, arg)
	}
```

I don't see a reason to identify the `enable-worker` and `enable-cloud-provider` keys here. It will require an extra change you have to make if you add any other flag without a value.

The intermediate `map[string]string` is a bit unnecessary too, the whole function could just simply be:

```go
func getCmdArgs(role string) args []string {
	args = append(args, role)
	args = append(args, "--data-dir", dataDir)
	if role == "server" {
		args = append(args, "--token-file", serverToken)

		if serverWorkerProfile != "default" {
			args = append(args, "--profile" , serverWorkerProfile)
		}

		if enableWorker {
			args = append(args, "--enable-worker")
		}

		if cfgFile != "" {
			args = append(args, "--config", cfgFile)
		}
	} else {
		args = append(args, "--cri-socket", criSocket)

		if workerProfile != "default" {
			args = append(args, "--profile", workerProfile)
		}

		if tokenFile != "" {
			args = append(args, "--token-file", tokenFile)
		} else {
			args = append(args, "--token-file", "REPLACEME")
		}

		if cloudProvider {
			args = append(args, "--enable-cloud-provider")
		}
	}

	if debug {
		args = append(args, "--debug")
	}

	return args
}
```

Shorter, simpler, less intermediate allocation, only uses anonymous string literals (=constants)

Reply:
@kke can you please re-review this file. I added requested changes.